graal-js/src/com.oracle.truffle.js.parser/src/com/oracle/truffle/js/parser/GraalJSTranslator.java

```diff
@@ -693,6 +695,12 @@ private JavaScriptNode toGeneratorBlockNode(AbstractBlockNode blockNode, BitSet
             }
             genBlock = returnsResult ? factory.createGeneratorExprBlock(chunks, readState, writeState) : factory.createGeneratorVoidBlock(chunks, readState, writeState);
         }
+        if (blockNode.hasTag(StandardTags.RootBodyTag.class)) {
+            genBlock.addRootBodyTag();
+        }
+        if (genBlock instanceof GeneratorExprBlockNode) {
+            tagHiddenExpression(genBlock);
+        }
```

Comment:
Could you move these two conditions to a dedicated method? Also, couldn't we perform some of the operations at node materialization time?

Reply:
I have updated the code. Now there is only one condition now, should I still put it in a dedicated method? 
It is more consistent to add the hidden expression tag in the parser here similar to other hidden expressions.

Reply:
OK, thanks. No need to add an extra method if it's just an extra `if` block